File: lua/difference-of-squares/difference-of-squares.lua
diff = {};

function diff.square_of_sums(n)
  local sum = 0

  for x = 1, n do
    sum = sum + x
  end

  return sum * sum
end

function diff.sum_of_squares(n)
  local sum = 0

  for x = 1, n do
    sum = sum + (x * x)
  end

  return sum
end

function diff.difference_of_squares(n)
  return diff.square_of_sums(n) - diff.sum_of_squares(n)
end

return diff;

File: php/difference-of-squares/difference-of-squares.php
<?php

function addSquare($sum, $x) {
  return $sum + $x ** 2;
}

function sumOfSquares($n)
{
  return array_reduce(range(1, $n), "addSquare", 0);
}

function squareOfSums($n)
{
  return array_sum(range(1, $n)) ** 2;
}

function difference($n)
{
  return squareOfSums($n) - sumOfSquares($n);
}

